extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamInfoItemExtractor.java

```diff
@@ -125,30 +128,27 @@ public String getUploaderName() throws ParsingException {
     @Override
     public String getUploaderUrl() throws ParsingException {
```

Comment:
In order to remove some checks and make more clear that there are many methods to extract the url which are tried one after the other, I'd suggest doing something like this:

```java
// try method one

if (url == null) {
    // try method two

    if (url == null) {
        // try method three

        if (url == null || url.isEmpty()) {
            // throw exception
        }
    }
}
```

Reply:
As in the following? Or what do you mean? I'm not sure if I think that's cleaner. @TobiGr: What do you think?

```
try {
    // Do whatever
} catch (Exception ignored) {}

if (url == null) {
    try {
        // Do whatever
    } catch (Exception ignored) {}

    if (url == null) {
        try {
            // Do whatever
        } catch (Exception ignored) {}

        if (url == null || url.isEmpty()) {
           throw new ParsingException("whatever");
        }
    }
}
```

Reply:
Well, the code above has less `url == null` checks when the first extraction method returns a valid result and is therefore "faster". On the other hand when the url is empty, we might not throw an Exception, but return an empty string. So we'd need something like this. 

```java
try {
    // Do whatever
} catch (Exception ignored) {}

if (url == null) {
    try {
        // Do whatever
    } catch (Exception ignored) {}

    if (url == null) {
        try {
            // Do whatever
        } catch (Exception ignored) {}
    }
}
if (url == null || url.isEmpty()) {
    throw new ParsingException("whatever");
}
return url;
```
However, I do not understand, why we only check once whether the url is empty. We should do this either every time or not at all. I suggest to remove this isEmpty check, because the JSON fields won't be empty. They either exist and contain a valid value or do not exist. That's why I'd use the following, which might be more efficient when having a valid result at the beginning. 

```java
try {
    // Do whatever
} catch (Exception ignored) {}

if (url == null) {
    try {
        // Do whatever
    } catch (Exception ignored) {}

    if (url == null) {
        try {
            // Do whatever
        } catch (Exception ignored) {}

        if (url == null) {
            throw new ParsingException("whatever");
        }
    }
}

return url;
```
However, these kind of changes won't speed up anything in a way a user can recognize that. If you want, you can change it, but maybe we should ask @Stypox first, if we understood correctly what he meant.

Reply:
In terms of speed up this does not actually do much, but I think the second snippet provided by @TobiGr is more readable, since the indentation makes it clear that if the first method succeeds, the function will return that result immediately. With the current implementation you have to read and process more lines in order to understand that if the first method succeeds nothing else happens (i.e. there is no post processing or other things) @TobiGr @wb9688